src/EloquentRepository.php

```diff
@@ -17,940 +17,953 @@
+     * @var int
+     */
+    protected $depth_restriction = 0;
+
+    /**
+     * Storage for query modifiers.
+     *
+     * @var array
+     */
+    private $modifiers = [];
+
+    /**
+     * The key name used in all queries.
+     *
+     * @var int
+     */
+    const KEY_NAME = 'id';
+
+    /**
+     * The glue for nested strings
+     *
+     * @var string
+     */
+    const GLUE = '.';
+
+    /**
+     * Set the model for an instance of this resource controller.
+     *
+     * @param string $model_class
+     * @return static
+     */
+    public function setModelClass($model_class)
+    {
+        if (!is_subclass_of($model_class, Model::class)) {
+            throw new \InvalidArgumentException('Specified model class must be an instance of ' . Model::class);
+        }
+
+        $this->model_class = $model_class;
+
+        return $this;
+    }
+
+    /**
+     * Get the model class.
+     *
+     * @return string
+     */
+    public function getModelClass()
+    {
+        return $this->model_class;
+    }
+
+    /**
+     * Set input manually.
+     *
+     * @param array $input
+     * @return static
+     */
+    public function setInput(array $input)
+    {
+        $this->input = $input;
+
+        return $this;
+    }
+
+    /**
+     * Get input.
+     *
+     * @return array
+     */
+    public function getInput()
+    {
+        return $this->input;
+    }
+
+    /**
+     * Set eager load manually.
+     *
+     * @param array $eager_loads
+     * @return static
+     */
+    public function setEagerLoads(array $eager_loads)
+    {
+        $this->eager_loads = $eager_loads;
+
+        return $this;
+    }
+
+    /**
+     * Get eager loads.
+     *
+     * @return array
+     */
+    public function getEagerLoads()
+    {
+        return $this->eager_loads;
+    }
+
+    /**
+     * Get the eager load depth property.
+     *
+     * @return int
+     */
+    public function getDepthRestriction()
+    {
+        return $this->depth_restriction;
+    }
+
+    /**
+     * Set the eager load depth property.
+     * This will limit how deep relationships can be included.
+     *
+     * @param int $depth
+     *
+     * @return $this
+     */
+    public function setDepthRestriction($depth)
+    {
+        $this->depth_restriction = $depth;
+
+        return $this;
+    }
+
+    /**
+     * Set filters manually.
+     *
+     * @param array $filters
+     * @return static
+     */
+    public function setFilters(array $filters)
+    {
+        $this->filters = $filters;
+
+        return $this;
+    }
+
+    /**
+     * Get filters.
+     *
+     * @return array
+     */
+    public function getFilters()
+    {
+        return $this->filters;
+    }
+
+    /**
+     * Add a filter to already existing filters without overwriting them.
+     *
+     * @param array $filters
+     * @return static
+     */
+    public function addFilters(array $filters)
+    {
+        $this->filters = array_merge($this->filters, $filters);
```

Comment:
This works well!

But say we also wanted to support adding a single filter:
```php
public function addFilter($key, $filter) {}
```

There's an implementation of this that supports both `addFilter` and `addFilters`, we might as well add that support in while we're at it.